### DISCUSSION TITLE
Can I simulate max_concurrency and batch_size params in event_source_mapping using aws_lambda_simple and sqs mocks?

### DISCUSSION THREAD

**Discussion by rishacha** [spulec/moto#7325] [Q&A] [answered]
I have a scenario where I'm using an architecture like this - 

1. I have a FIFO queue
2. I pushing objects to the FIFO queue
3. I have setup an event source mapping to aws_lambda
4. I want to test out the batch_size params, and max_concurrency params

How do I go about simulating/testing this scenario using a custom function and having it invoked multiple times with the event_source_mapping?

My idea is to use - aws_lambda_simple to do the job. 

Would this work?

**Comment by bblommers** [comment, marked as answer]
Hi @rishacha! Moto isn't quite advanced enough to support your test case, I'm afraid. Invoking a function via an EventSource is done syncronously, so the `batch_size` will realistically always be 1. The `MaximumConcurrency`-argument isn't taken into account at all, for the same reason.

Maybe LocalStack may be a better fit? I don't know for sure whether they support it, but they tend to be better at emulating the real behaviour when it comes to AWSLambda Functions.

**Reply by UseTheFork** [reply]
Hi! Can I ask a follow-up? What's the best mechanism to use to log or confirm that an SQS message triggered a Lambda function? I have the Lambda and SQS loaded, but I can't figure out if my Lambda is triggering. Thanks in advance!

**Reply by bblommers** [reply]
All Lambda logs are written to CloudWatchLogs, so I typically check those if I want to see if a Lambda was triggered.

We use this method internally, if you want ~to copy/paste~ some inspiration:
https://github.com/getmoto/moto/blob/e1ddea8a0257bd069c2c1e564327d8543e4eaed5/tests/test_awslambda/utilities.py#L202

**Reply by UseTheFork** [reply]
I figured this out. Everything works as expected! I just didn't see the warning in the docs about the url in a moto server needed to be updated to point back to moto not aws.

One followup question 😀
I use the "moto-api/" a lot. But one downside is if I load a larger lambda function the page hangs for a while since it's trying to display a base64 encoded 30meg file. Is there a way to turn off it displaying the contents of the lambda function? 

I'm guessing no based on the current contents of https://github.com/getmoto/moto/blob/c2d61ae6d6e5bfac7b51c8e86059f67e28cb5b0a/moto/moto_server/templates/dashboard.html

**Reply by bblommers** [reply]
Ah, good to hear that it works!

The dashboard is indeed fairly straightforward - there's really no options to limit or filter by attributes. PR's are welcome if you want to improve this! 🙂
